api-server/web-server/src/api/v1.rs

```diff
@@ -315,8 +317,77 @@ async fn get_transaction(
         ))
 }
 
+pub async fn feerate<T: ApiServerStorage>(
+    Query(params): Query<BTreeMap<String, String>>,
+    State(state): State<ApiServerWebServerState<Arc<T>, Arc<impl TxSubmitClient>>>,
+) -> Result<impl IntoResponse, ApiServerWebServerError> {
+    const REFRESH_INTERVAL_SEC: u64 = 30;
+    const IN_TOP_X_MB: &str = "in_top_x_mb";
+    const DEFAULT_IN_TOP_X_MB: usize = 5;
+    let in_top_x_mb = params
+        .get(IN_TOP_X_MB)
+        .map(|str| usize::from_str(str))
+        .transpose()
+        .map_err(|_| ApiServerWebServerClientError::InvalidInTopX)?
+        .unwrap_or(DEFAULT_IN_TOP_X_MB);
+
+    let feerate_points = &state.cached_values.feerate_points;
+
+    let feerate_points: BTreeMap<_, _> = {
+        let current_time = get_time();
```

Comment:
Please use TimeGetter, and don't use direct time access in production code. Then to test this feature, you can simulate time. See how it's done in other crates, such as ChainState.